### PR TITLE
Add `write_content`.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/file.toit [hunk 1]
```diff
@@ -45,61 +46,126 @@ class Stream implements Reader:
 
   constructor.internal_ .fd_:
 
-  // Open a file for reading.
-  constructor.for_read path:
+  /**
```
Comment (floitsch) [inline src/file.toit:49]: Most functions only have their toitdocs changed, and I often changed the "filename"/"name" to "path".
Also added some types when I could.

File: src/file.toit [hunk 2]
```diff
@@ -45,61 +46,126 @@ class Stream implements Reader:
...
+  Opens the file at $path for writing.
 
-  constructor name flags:
+  If the file does not exist, it is created.  If it exists, it is truncated.
+  Uses the given $permissions, modified by the current umask, to set the
+    permissions of the file.
+
+  Ignored if the file already exists.
+  */
+  constructor.for_write path/string --permissions/int=((6 << 6) | (6 << 3) | 6):
```
Comment (floitsch) [inline src/file.toit:64]: Added a `--permissions`.

File: src/file.toit [hunk 3]
```diff
@@ -45,61 +46,126 @@ class Stream implements Reader:
...
+read_contents name:
+  return read_content name
 
-// Takes an open file descriptor and determines if it represents a file
-// as opposed to a socket or a pipe.
-is_open_file_ fd:
-  #primitive.file.is_open_file
+/**
+Reads the content of a file.
+The file must not change while it is read into memory.
```
Comment (floitsch) [inline src/file.toit:128]: read_content function only moved.

File: src/file.toit [hunk 4]
```diff
@@ -45,61 +46,126 @@ class Stream implements Reader:
...
+
+/**
+Writes the given $content to a file of the given $path.
+The file must not change while it is read into memory.
+
+If $permissions is provided uses it to set the permissions of the file.
+The $permissions are only used if the file is created, and not if it is
+  overwritten.
+*/
+write_content content --path/string --permissions/int?=null -> none:
```
Comment (floitsch) [inline src/file.toit:162]: New function.

File: src/file.toit [hunk 5]
```diff
@@ -120,6 +186,21 @@ size name:
   if stat[ST_TYPE] != REGULAR_FILE: throw "INVALID_ARGUMENT"
   return stat[ST_SIZE]
 
+// Returns a file descriptor.  Only for use on actual files, not pipes,
```
Comment (floitsch) [inline src/file.toit:189]: Only moved.

**Review by erikcorry — APPROVED** [review]

**Pull request merged**